kafka/channel/pkg/dispatcher/dispatcher.go

```diff
@@ -303,33 +320,80 @@ func (d *KafkaDispatcher) getChannelReferenceFromHost(host string) (eventingchan
 	return cr, nil
 }
 
-func fromKafkaMessage(kafkaMessage *sarama.ConsumerMessage) *contribchannels.Message {
-	headers := make(map[string]string, len(kafkaMessage.Headers))
+func fromKafkaMessage(ctx context.Context, kafkaMessage *sarama.ConsumerMessage) *cloudevents.Event {
+	var event = &cloudevents.Event{}
 	for _, header := range kafkaMessage.Headers {
-		headers[string(header.Key)] = string(header.Value)
-	}
-	message := contribchannels.Message{
-		Headers: headers,
-		Payload: kafkaMessage.Value,
+		h := string(header.Key)
+		v := string(header.Value)
+		logging.FromContext(ctx).Debugf("key: %s, value %s", h, v)
+		switch h {
+		case "ce_datacontenttype":
+			event.SetDataContentType(v)
+		case "ce_specversion":
+			event.SetSpecVersion(v)
+		case "ce_type":
+			event.SetType(v)
+		case "ce_source":
+			event.SetSource(v)
+		case "ce_id":
+			event.SetID(v)
+		case "ce_time":
+			t, _ := time.Parse(time.RFC3339, v)
+			event.SetTime(t)
+		case "ce_subject":
+			event.SetSubject(v)
+		case "ce_dataschema":
+			event.SetDataSchema(v)
+		default:
+			// Extensions
+			event.SetExtension(strings.ToLower(h), strings.ToLower(v))
```

Comment:
This might break if the header is non-alphanumeric in the case of 1.0. 
I had to do some ugly check for our PubSub source here: https://github.com/google/knative-gcp/blob/master/pkg/pubsub/adapter/converters/pubsub.go#L54
you might need something similar..